concreate/descriptor.py

```diff
@@ -109,6 +114,58 @@ def _process_labels(self):
             self.descriptor['labels'].append({'name': 'description',
                                               'value': self.descriptor['description']})
 
+class Dependency(object):
+    def new(dependency):
+        if 'git' in dependency:
```

Comment:
If Dependency has both and 'git' and 'directory' defined, git will be used here, is it correct for overrides? I would expect   it vice versa for local development story.

Reply:
Theoretically, it should error if both are present.  As far as merging goes, the merge is done after processing, so the keys should wipe each other out, i.e. cct_module=GitDependency(...) <= cct_module=DirectoryDependency(...), there shouldn't be any overlap, in theory anyway (python isn't my strong suit).